### ISSUE TITLE
Change and unify where pytest output is saved

### ISSUE THREAD

**Issue by rickecon** [PSLmodels/OG-Core#770]
Currently, when the full set of tests from the `pytest` package is run, the output is saved in multiple places, some of which are in the main directory. The following is the list of `pytest` saved output from a recent full run of the unit tests.
```
OUTPUT_BASELINE/
ogcore/tests/OUTPUT_REFORM_/
ogcore/tests/TPI/
ogcore/tests/test_out.pkl
output.csv
output.json
output.tex
output.xlsx
```
I propose that all testing output be saved in a folder in the `OG-Core/ogcore/tests/` directory that has a name based off of the timestamp when the tests were begun. For example, the directory for a test run today at 1:50p local computer time might be `2022-01-05-1350`. The status check of the saved output would be the following.
```
ogcore/tests/2022-01-05-1350/OUTPUT_BASELINE/
ogcore/tests/2022-01-05-1350/OUTPUT_REFORM_/
ogcore/tests/2022-01-05-1350/TPI/
ogcore/tests/2022-01-05-1350/test_out.pkl
ogcore/tests/2022-01-05-1350/output.csv
ogcore/tests/2022-01-05-1350/output.json
ogcore/tests/2022-01-05-1350/output.tex
ogcore/tests/2022-01-05-1350/output.xlsx
```
This would make testing run output easy to save, archive, and erase.

cc: @jdebacker

**Comment by jdebacker** [comment]
@rickecon Thanks for pointing this out.  A number of files (and perhaps all) created are of no value afterward.  We could use the [`pytest` `tmpdir`](https://docs.pytest.org/en/6.2.x/tmpdir.html) fixture to save these to a temporary directory.  This is done with some other tests in OG-Core, such as [test_parameter_plots.test_plot_mort_rates_save_fig](https://github.com/PSLmodels/OG-Core/blob/f0e241690ce64616843ab82ebb4e5fbe038ec9ec/ogcore/tests/test_parameter_plots.py#L46

**Comment by rickecon** [comment]
This Issue is resolved by PR #775.
